homeassistant/components/media_player/emby.py

```diff
@@ -4,252 +4,227 @@
-            return
-
-        new_emby_clients = []
-        for device in devices:
-            if device['DeviceId'] == embyserver.unique_id:
-                break
-
-            if device['DeviceId'] not in emby_clients:
-                _LOGGER.debug('New Emby DeviceID: %s. Adding to Clients.',
-                              device['DeviceId'])
-                new_client = EmbyClient(embyserver, device, emby_sessions,
-                                        update_devices, update_sessions)
-                emby_clients[device['DeviceId']] = new_client
-                new_emby_clients.append(new_client)
-            else:
-                emby_clients[device['DeviceId']].set_device(device)
-
-        if new_emby_clients:
-            add_devices_callback(new_emby_clients)
-
-    @Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
-    def update_sessions():
-        """Update the sessions objects."""
-        sessions = embyserver.get_sessions()
-        if sessions is None:
-            _LOGGER.error('Error listing Emby sessions')
-            return
+    from pyemby import EmbyServer
+
+    host = config.get(CONF_HOST)
+    key = config.get(CONF_API_KEY)
+    port = config.get(CONF_PORT)
+    ssl = config.get(CONF_SSL)
+    auto_hide = config.get(CONF_AUTO_HIDE)
+
+    _LOGGER.debug('Setting up Emby server at: %s', host)
+
+    emby = EmbyServer(host, key, port, ssl, hass.loop)
+
+    active_emby_devices = {}
+    hidden_emby_devices = {}
+
+    @callback
+    def device_update_callback(data):
+        """Callback for when devices are added to emby."""
+        new_devices = []
+        active_devices = []
+        for dev_id in emby.devices:
+            active_devices.append(dev_id)
+            if dev_id not in active_emby_devices and \
+                    dev_id not in hidden_emby_devices:
+                new = EmbyDevice(emby, dev_id)
+                active_emby_devices[dev_id] = new
+                new_devices.append(new)
+
+            elif dev_id in hidden_emby_devices:
+                if emby.devices[dev_id].state != 'Off':
+                    add = hidden_emby_devices.pop(dev_id)
+                    active_emby_devices[dev_id] = add
+                    _LOGGER.debug("Showing %s, item: %s", dev_id, add)
+                    add.set_hidden(False)
+
+        if new_devices:
+            _LOGGER.debug("Adding new devices to HASS: %s", new_devices)
+            async_add_devices(new_devices, update_before_add=True)
+
+    @callback
+    def device_removal_callback(data):
+        """Callback for when devices are removed from emby."""
+        if auto_hide:
+            if data in active_emby_devices:
+                rem = active_emby_devices.pop(data)
+                hidden_emby_devices[data] = rem
+                _LOGGER.debug("Hiding %s, item: %s", data, rem)
+                rem.set_hidden(True)
+
+    def start_emby(event):
+        """Start emby connection."""
+        emby.start()
```

Comment:
The start method of the pyEmby library runs a register coroutine in the loop.  I can bypass the start routine and likely just call the register coroutine if you'd like.